core/src/main/scala/org/apache/spark/NewAccumulator.scala

```diff
@@ -209,9 +210,9 @@ private[spark] object AccumulatorContext {
    */
   def newId(): Long = nextId.getAndIncrement
 
-  def numAccums: Int = synchronized(originals.size)
+  def numAccums: Int = originals.size
 
-  def accumIds: Set[Long] = synchronized(originals.keySet().asScala.toSet)
+  def accumIds: Set[Long] = originals.keySet().asScala.toSet
```

Comment:
It still needs to be synchronized as it this iterates over the map, and that does not happen atomically. Even if you argue it doesn't happen to occur now given its usage, it's more correct.